src/cmds/service.c

```diff
@@ -11,15 +11,69 @@
 #include <string.h>
 #include <unistd.h>
 #include <stdlib.h>
+#include <mem/misc/pool.h>
+#include <util/dlist.h>
+#include <sys/types.h>
+#include <signal.h>
 
-static void srv_print_services(void) {
-	printf("PRINT LIST OF SERVICES IS NOT SUPPORTED YET\n");
+#define len 32
+
+struct service_list {
+	struct dlist_head dlist_item;
+	char service_name[len];
+	pid_t pid;
+};
+
+POOL_DEF(pool, struct service_list, sizeof(struct service_list));		 
+
+static DLIST_DEFINE(head);
+
+static void add_element_in_list(const char *name, pid_t service_pid) {
+  	struct service_list *new_element;
+
+  	new_element = pool_alloc(&pool);
+   
+  	strncpy(new_element->service_name,name,len);
```

Comment:
Here you dereference the pointer, but it can be null (`new_element` can be NULL if pool is full), which will cause system crash, and that's not what we want. It's better to return error code and/or print appropriate error message.